src/cli_parser.cpp

```diff
@@ -508,8 +516,26 @@ void CLIParser::prepareGraph(ServerSettingsImpl& serverSettings, HFSettingsImpl&
 
         if (result->count("overwrite_models"))
             hfSettings.overwriteModels = result->operator[]("overwrite_models").as<bool>();
-        if (result->count("source_model"))
+        if (result->count("source_model")) {
             hfSettings.sourceModel = result->operator[]("source_model").as<std::string>();
+            // FIXME: Currently we use git clone only for OpenVINO, we will change this method of detection to parsing model files
+            if (!startsWith(toLower(serverSettings.hfSettings.sourceModel), toLower("OpenVINO/"))) {
+                std::cout << "model" << serverSettings.hfSettings.sourceModel << std::endl;
+                hfSettings.downloadType = OPTIMUM_CLI_DOWNLOAD;
+            }
+        }
+
+        if (result->count("precision") && hfSettings.downloadType == GIT_CLONE_DOWNLOAD) {
+            throw std::logic_error("--precision parameter unsupported for Openvino huggingface organization models.");
+        }
+        if (result->count("extra_quantization_params") && hfSettings.downloadType == GIT_CLONE_DOWNLOAD) {
+            throw std::logic_error("--extra_quantization_params parameter unsupported for Openvino huggingface organization models.");
+        }
+        std::cout << "validated pull" <<std::endl;
+        if (result->count("precision"))
+            hfSettings.precision = result->operator[]("precision").as<std::string>();
```

Comment:
We are relying on precision string correctness validation from optimum-cli correct? Is error message helpful in that case/